src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseChecklistService.cs

```diff
@@ -0,0 +1,196 @@
+        {
+            _persistenceHelper = persistenceHelper;
+            _tableStorageService = tableStorageService;
+            _userService = userService;
+            _metaGuidanceService = metaGuidanceService;
+            _fileRepository = fileRepository;
+            _footnoteRepository = footnoteRepository;
+            _dataBlockService = dataBlockService;
+        }
+
+        public async Task<Either<ActionResult, ReleaseChecklistViewModel>> GetChecklist(Guid releaseId)
+        {
+            return await _persistenceHelper.CheckEntityExists<Release>(releaseId, HydrateReleaseForChecklist)
+                .OnSuccess(_userService.CheckCanViewRelease)
+                .OnSuccess(
+                    async release => new ReleaseChecklistViewModel(
+                        await GetErrors(release),
+                        await GetWarnings(release)
+                    )
+                );
+        }
+
+        public static IQueryable<Release> HydrateReleaseForChecklist(IQueryable<Release> query)
+        {
+            return query.Include(r => r.Publication)
+                .ThenInclude(p => p.Methodology)
+                .Include(r => r.Updates);
+        }
+
+        public async Task<List<ReleaseChecklistIssue>> GetErrors(Release release)
+        {
+            var errors = new List<ReleaseChecklistIssue>();
+
+            if (await HasDataFilesUploading(release))
+            {
+                errors.Add(new ReleaseChecklistIssue(ValidationErrorMessages.DataFileImportsMustBeCompleted));
+            }
+
+            var replacementDataFiles = await _fileRepository.ListReplacementDataFiles(release.Id);
+
+            if (replacementDataFiles.Any())
+            {
+                errors.Add(new ReleaseChecklistIssue(ValidationErrorMessages.DataFileReplacementsMustBeCompleted));
+            }
+
+            if (release.Publication.Methodology != null
+                && release.Publication.Methodology.Status != MethodologyStatus.Approved)
+            {
+                errors.Add(new MethodologyMustBeApprovedError(release.Publication.Methodology.Id));
+            }
+
+            var isMetaGuidanceValid = await _metaGuidanceService.Validate(release.Id);
+
+            if (isMetaGuidanceValid.IsLeft)
+            {
+                errors.Add(new ReleaseChecklistIssue(ValidationErrorMessages.PublicMetaGuidanceRequired));
+            }
+
+            if (release.PreReleaseAccessList.IsNullOrEmpty())
+            {
+                errors.Add(new ReleaseChecklistIssue(ValidationErrorMessages.PublicPreReleaseAccessListRequired));
+            }
+
+            if (release.Amendment && release.Updates.All(update => update.ReleaseId != release.Id))
+            {
+                errors.Add(new ReleaseChecklistIssue(ValidationErrorMessages.ReleaseNoteRequired));
+            }
+
+            return errors;
+        }
+
+        private async Task<bool> HasDataFilesUploading(Release release)
```

Comment:
Yeh it would, but I would still make the user action these failed or cancelled files in the UI. Don't think it would be a good idea  to assume that the release is safe to publish if we have any imports in those states!